winml/lib/Api.Ort/OnnxruntimeErrors.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+#include "pch.h"
+#include "core/providers/winml/winml_provider_factory.h"
+
+#ifdef _WIN32
+inline HRESULT OrtErrorCodeToHRESULT(OrtErrorCode status) noexcept {
+  switch (status) {
+    case OrtErrorCode::ORT_OK:
+      return S_OK;
+    case OrtErrorCode::ORT_FAIL:
+      return E_FAIL;
+    case OrtErrorCode::ORT_INVALID_ARGUMENT:
+      return E_INVALIDARG;
+    case OrtErrorCode::ORT_NO_SUCHFILE:
+      return __HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND);
+    case OrtErrorCode::ORT_NO_MODEL:
+      return __HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND);
+    case OrtErrorCode::ORT_ENGINE_ERROR:
+      return E_FAIL;
+    case OrtErrorCode::ORT_RUNTIME_EXCEPTION:
+      return E_FAIL;
+    case OrtErrorCode::ORT_INVALID_PROTOBUF:
+      return __HRESULT_FROM_WIN32(ERROR_FILE_CORRUPT);
+    case OrtErrorCode::ORT_MODEL_LOADED:
+      return __HRESULT_FROM_WIN32(ERROR_INTERNAL_ERROR);
+    case OrtErrorCode::ORT_NOT_IMPLEMENTED:
+      return E_NOTIMPL;
+    case OrtErrorCode::ORT_INVALID_GRAPH:
+      return __HRESULT_FROM_WIN32(ERROR_FILE_CORRUPT);
+    case OrtErrorCode::ORT_EP_FAIL:
+      return __HRESULT_FROM_WIN32(ERROR_INTERNAL_ERROR);
+    default:
+      return E_FAIL;
+  }
+}
+#endif
+
+#define RETURN_HR_IF_WINMLA_API_FAIL_MSG(status, ort_api)                                                      \
```

Comment:
>RETURN_HR_IF_WINMLA_API_FAIL_MSG [](start = 8, length = 32)

Please dont call it winmla - the statuses are just ORT statuses. Nothing adapter specific about them.

Reply:
should probably have IF_NOT_OK_MSG as suffix.

---
In reply to: [369137077](https://github.com/microsoft/onnxruntime/pull/2869#discussion_r369137077) [](ancestors = 369137077)